java/analyzer/src/main/java/org/wikidata/analyzer/WikidataAnalyzer.java

```diff
@@ -63,21 +113,28 @@ public static void main(String[] args) throws IOException {
         System.out.println("Processed!");
         System.out.println("Memory Usage (MB): " + Runtime.getRuntime().totalMemory() / 1024 / 1024);
 
-        // Create all output files
-        System.out.println("Writing map wdlabel.json");
-        File mapLabelFile = new File( dataDir.getAbsolutePath() + File.separator + "wdlabel.json");
-        BufferedWriter mapLabelWriter = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(mapLabelFile)));
-        mapGeoData.writeJSONString(mapLabelWriter);
-        mapLabelWriter.close();
-        System.out.println("Writing map graph.json");
-        File mapGraphFile = new File( dataDir.getAbsolutePath() + File.separator + "graph.json");
-        BufferedWriter mapGraphWriter = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(mapGraphFile)));
-        mapGraphData.writeJSONString(mapGraphWriter);
-        mapGraphWriter.close();
+        // Map
+        if( processors.contains( "Map" ) ) {
+            System.out.println("Writing map wdlabel.json");
+            File mapLabelFile = new File( dataDir.getAbsolutePath() + File.separator + "wdlabel.json");
+            BufferedWriter mapLabelWriter = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(mapLabelFile)));
+            mapGeoData.writeJSONString(mapLabelWriter);
+            mapLabelWriter.close();
+            System.out.println("Writing map graph.json");
+            File mapGraphFile = new File( dataDir.getAbsolutePath() + File.separator + "graph.json");
+            BufferedWriter mapGraphWriter = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(mapGraphFile)));
+            mapGraphData.writeJSONString(mapGraphWriter);
+            mapGraphWriter.close();
+        }
```

Comment:
This is a rather long method, over 50 lines. It is also full of low level details such as strings to output and contains things such as construction logic. Splitting this up a bit would be nice